scanspec/regions.py

```diff
@@ -242,6 +242,55 @@ def mask(self, points: AxesPoints) -> np.ndarray:
         return mask_x & mask_y
 
 
+@dataclass
+class Polygon(Region):
```

Comment:
Do any users ever want regular polygons or is the only use case me wanting to test it without working out vertex coordinates?

Reply:
@EmmaCD what do you mean by regular polygons? Like a hexagon? If so, then I don't think anyone wants that. This is for "my sample is a funny shape, draw a polygon around it an only scan that bit"